pkgs/by-name/re/resnap/package.nix

```
@@ -0,0 +1,54 @@
+{
+  lib,
+  stdenvNoCC,
+  fetchFromGitHub,
+  ffmpeg,
+  feh,
+  imagemagick_light,
+  lz4,
+}:
+let
+  name = "reSnap";
+in
+stdenvNoCC.mkDerivation (finalAttrs: {
+  pname = lib.strings.toLower name;
+  version = "2.5.2";
+
+  src = fetchFromGitHub {
+    owner = "cloudsftp";
+    repo = name;
+    rev = "v${finalAttrs.version}";
+    hash = "sha256-thVyf1gTDPLQVtZKoWL7SGiWI++ICWqmF/Ar57I3WP8=";
+  };
+
+  postFixup = ''
+    substituteInPlace $out/bin/reSnap \
+      --replace-fail "\$0" ${name}
+  '';
+
+  installPhase = ''
+    runHook preInstall
+
+    mkdir -p $out/bin
+    echo PATH="\$PATH:${
+      lib.strings.makeBinPath [
+        ffmpeg
+        feh
+        imagemagick_light
+        lz4
+      ]
+    }" > $out/bin/reSnap
+    cat ./reSnap.sh >> $out/bin/reSnap
+    chmod +x $out/bin/reSnap
+
+    runHook postInstall
+  '';
```

We can use `runtimeInputs` or wrap the executable instead of doing this.

```suggestion
   installPhase = ''
    runHook preInstall

    install -D ./reSnap.sh $out/bin/reSnap

    runHook postInstall
  '';
```